geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloDataStoreFactory.scala

```diff
@@ -121,7 +121,10 @@ class AccumuloDataStoreFactory extends DataStoreFactorySpi {
         .map(FeatureEncoding.withName)
         .getOrElse(FeatureEncoding.AVRO)
 
-    val collectStats = !useMock && Try(statsParam.lookUp(params).asInstanceOf[java.lang.Boolean] == true).getOrElse(true)
+    val collectStats = !useMock &&
+        Option(statsParam.lookUp(params).asInstanceOf[java.lang.Boolean]).forall(_ == true)
+    val caching =
+      Option(cachingParam.lookUp(params).asInstanceOf[java.lang.Boolean]).getOrElse(!useMock) == true
```

Comment:
hmm, actually it does seem to be a java/scala issue... i don't see another option that is clearer/cleaner than this.